### ISSUE TITLE
Deprecate magic train untrain methods

### ISSUE THREAD

**Issue by ibnesayeed** [jekyll/classifier-reborn#140]
Before releasing `2.2.0` magical `train_*` and `untrain_*` style methods should be deprecated.

* [ ] Add deprecation warning
* [ ] Add tests to not use these methods
* [ ] Update documentation to not use these methods

**Comment by ibnesayeed** [comment]
It would also be good to have the official development Docker image available (#101) before the release of `2.2.0` so that it can be documented and advertised with this rather significant release.

**Comment by ibnesayeed** [comment]
Perhaps we can also include basic validation #71 as part of `2.2.0` release.

**Comment by Ch4s3** [comment]
Sounds good. I have a working branch for the first two items. I'll push it up later.

**Comment by ibnesayeed** [comment]
I can take care of the documentation, later tonight.

**Comment by ibnesayeed** [comment]
PR #141 takes care of the documentation.

**Comment by ibnesayeed** [comment]
@Ch4s3 a note for you. I have seen `test_external_classifying` of `lsi_test.rb` using magic methods. Just make sure to take care of all the places where Bayes is being called with magic methods.

**Comment by Ch4s3** [comment]
Yeah, that's on the list. I'll try to devote some time to this tomorrow or Monday.

**Comment by Ch4s3** [comment]
I'll try to get back to this asap

**Comment by ibnesayeed** [comment]
@Ch4s3 please take care of the second point on the original post of this ticket before releasing 2.2.0 as per #164. This should be pretty easy and quick to deal with.

**Comment by Ch4s3** [comment]
@ibnesayeed will do.
